client/components/payment-activity/payment-data-tile.tsx

```diff
@@ -41,6 +42,10 @@ interface PaymentDataTileProps {
 	 * Optional hover link to view report.
 	 */
 	reportLink?: string;
+	/**
+	 * Optional click handler for the tile.
+	 */
+	handleReportLinkClick?: () => void;
```

Comment:
Is our `PaymentDataTile` abstraction helping here? If we expose a prop for every detail of it, we might be better off just using `Link` directly in the outer markup. 

Alternatively, if we are ok using the same event name for all tiles e.g. `wcpay_overview_payment_tile_click`, the component could abstract away the tracking behaviour nicely. It could use an existing prop or expose a `tracksEventSource` prop, which would avoid complex callback interface.

Reply:
Explained in the earlier comment. 

https://github.com/Automattic/woocommerce-payments/pull/8687#discussion_r1580961883 

Thanks!